### DISCUSSION TITLE
HasManyThrough not working after Laravel 11 upgrade

### DISCUSSION THREAD

**Discussion by thomasschiet** [larastan/larastan#2070] [Q&A] [answered]
Hi all,

I recently upgraded from Laravel 10 to Laravel 11. Oddly enough I cannot get a HasManyThrough relationship to work. I have implemented the relationship as follows
```php
    /** @return HasManyThrough<Deployment, Environment, Project> */
    public function deployments(): HasManyThrough
    {
        return $this->hasManyThrough(Deployment::class, Environment::class);
    }
```
Which should be correct judging by [Laravel source code](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Relations/HasOneThrough.php) and the empty [Larastan stub](https://github.com/larastan/larastan/blob/2.x/stubs/11.0.0/HasManyThrough.stub). 

The error I get is
```
Generic type Illuminate\Database\Eloquent\Relations\HasManyThrough<App\Deployment, App\Environment, App\Project>
 in PHPDoc tag @return specifies 3 template types,
 but class Illuminate\Database\Eloquent\Relations\HasManyThrough supports only 1: TRelatedModel  
```

This leads me to believe the incorrect sub is used: https://github.com/larastan/larastan/blob/2.x/stubs/11.0.0/HasManyThrough.stub.

I have tried clearing PHPStan cache but to no avail.

I am using Laravel v11.27.2 and Larastan v2.9.8.

**Comment by szepeviktor** [comment]
Thank you for your report.

https://github.com/larastan/larastan/blob/d009b01aef43c7c4072113fcfc6fa767762691d5/stubs/common/HasManyThrough.stub#L6
Try `/** @return HasManyThrough<Deployment> */`

**Comment by canvural** [comment, marked as answer]
Hi,

Actually those changes are not yet released. Can you test it with the `2.x` branch?
